datalad_next/patches/push_to_export_remote.py

```diff
@@ -0,0 +1,240 @@
+
+        lgr.debug("Exporting HEAD to a remote with exporttree == yes")
+
+        if ds.config.getbool('remote.{}'.format(target), 'annex-ignore', False):
+            lgr.debug(
+                "Target '%s' is set to annex-ignore, exclude from data-export.",
+                target)
+            return
+
+        if force not in ("all", "export"):
+            export_entry = _get_export_log_entry(repo, target_uuid)
+            if export_entry:
+                if export_entry["source-annex-uuid"] != repo.uuid:
+                    yield dict(
+                        **res_kwargs,
+                        status="error",
+                        message=f"refuse to export to {target}, because the "
+                                f"last known export came from another repo "
+                                f"({export_entry['source-annex-uuid']}). Use "
+                                f"--force=export to enforce the export anyway.")
+                    return
+                if not _is_valid_treeish(repo, export_entry):
+                    yield dict(
+                        **res_kwargs,
+                        status="error",
+                        message=f"refuse to export to {target}, because the "
+                                f"current state is not a fast-forward of the "
+                                f"last known exported state. Use "
+                                f"--force=export to enforce the export anyway.")
+                    return
+
+        credentials = _get_credentials(ds, remote_info)
+
+        # If we have credentials, check whether we require an environment patch
+        env_patch = {}
+        remote_type = remote_info.get("type")
+        if credentials and needs_specialremote_credential_envpatch(remote_type):
+            env_patch = get_specialremote_credential_envpatch(
+                remote_type,
+                credentials)
+
+        res_kwargs['target'] = target
+
+        with patch.dict('os.environ', env_patch):
+            try:
+                repo.call_git(["annex", "export", "HEAD", "--to", target])
+            except CommandError as cmd_error:
+                ce = CapturedException(cmd_error)
+                yield {
+                    **res_kwargs,
+                    "action": "copy",
+                    "status": "error",
+                    "message": str(ce),
+                    "exception": ce
+                }
+                return
+
+        yield {
+            **res_kwargs,
+            "action": "copy",
+            "status": "ok",
+        }
+
+    else:
+        yield from push._push_data(
+            ds,
+            target,
+            content,
+            data,
+            force,
+            jobs,
+            res_kwargs.copy(),
+            got_path_arg=got_path_arg,
+        )
+
+
+lgr.debug("Patching datalad.core.distributed.push._transfer_data")
+push._transfer_data = _transfer_data
```

Comment:
I'd say that we also need to amend the main `Push.__doc__` and announce the additional capabilities. Maybe just append

```suggestion
push._transfer_data = _transfer_data
push.Push.__doc__ += """\


Features added by the datalad-next extension

Transparently transfer data to git-annex special remotes in export-mode (`exporttree=yes`), whenever ...
"""
```